doc/documentation.md

```diff
@@ -0,0 +1,739 @@
+        leading: Showcase(
+          key: _one,
+          title: 'Menu',
+          description: 'Click here to see menu options',
+          child: Icon(Icons.menu),
+        ),
+      ),
+      floatingActionButton: Showcase(
+        key: _two,
+        title: 'Add',
+        description: 'Click here to add new items',
+        child: FloatingActionButton(
+          onPressed: () {},
+          child: Icon(Icons.add),
+        ),
+      ),
+      body: Center(
+        child: Text('ShowCase Example'),
+      ),
+    );
+  }
+  
+  @override
+  void dispose() {
+    // Unregister the showcase view
+    ShowcaseView.get().unregister();
+    super.dispose();
+  }
+}
+```
+
+# Advanced Usage
+
+This guide covers more advance features and customizations of the ShowCaseView package.
+
+## Auto Play
+
+Enable auto play to automatically advance through showcases:
+
+```dart
+ShowcaseView.register(
+  autoPlay: true,
+  autoPlayDelay: Duration(milliseconds: 3000),
+  enableAutoPlayLock: true,
+)
+```
+
+## Auto Scrolling
+
+Enable auto scrolling to automatically bring off-screen showcase widgets into view:
+
+```dart
+ShowcaseView.register(
+  enableAutoScroll: true,
+  scrollDuration: Duration(milliseconds: 500),
+)
+```
+
+> **Note:** Auto-scroll does not work with multi-showcase, and in order to scroll to a widget it 
+> needs to be attached with widget tree.
+>
+> If you are using a scrollview that renders widgets on demand like ListView.builder, it is 
+> possible that the widget to be showcased is not attached to the widget tree leaving flutter 
+> unable to scroll to that widget.
+>
+> In such cases, you can assign a ScrollController to that scrollview and manually scroll to 
+> the position where showcase widget gets rendered. You can add code for that in the onStart 
+> parameter of the ShowcaseView.register as shown below.
+
+Example:
+```dart
+final _controller = ScrollController();
+
+ShowcaseView.register(
+  onStart: (index, key) {
+    if(index == 0) {
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        // Scroll to approximate position of the showcase widget
+        _controller.jumpTo(1000);
+      });
+    }
+  },
+),
+
+// In your ListView:
+ListView.builder(
+  controller: _controller,
+  itemCount: 100,
+  itemBuilder: (context, index) {
+    // Your list items
+  },
+)
+```
+
+## Multi-Showcase View
+
+To show multiple showcases simultaneously, use the same key for multiple showcase widgets:
+
+```dart
+// Both will be displayed at the same time
+Showcase(
+  key: _multiKey,
+  title: 'First Widget',
+  description: 'This is the first widget',
+  child: Icon(Icons.star),
+),
+
+Showcase(
+  key: _multiKey,
+  title: 'Second Widget',
+  description: 'This is the second widget',
+  child: Icon(Icons.favorite),
+),
+```
+
+> **Note:** Auto-scroll does not work with multi-showcase, and properties of the first 
+> initialized showcase are used for common settings like barrier tap and colors.
```

Comment:
Since duplicated keys are no longer allowed in the latest version of this library, this statement is no longer valid and should be removed.